packages/flutter/lib/src/painting/binding.dart

```diff
@@ -53,7 +53,20 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   /// See also:
   ///
   ///  * [ShaderWarmUp], the interface of how this warm up works.
-  static ShaderWarmUp shaderWarmUp = const DefaultShaderWarmUp();
+  static ShaderWarmUp get shaderWarmUp => _shaderWarmUp;
+  static void set shaderWarmUp(ShaderWarmUp value) {
+    assert(_instance == null,
+        'PaintingBinding.shaderWarmUp should only be set before the '
+        'PaintingBinding singleton instance is initialized.\n\n'
+        'Set it after init would not affect how shaders are warmed up.\n\n'
```

Comment:
small nit:

```suggestion
        'Setting it after init does not affect how shaders are warmed up.\n\n'
```

"Set" here makes it sound like a command so it's confusing as is, because I think this sentence is really talking about the times when this is set after init and not telling the reader to set it after init. ["Setting" changes it to a gerund and the subject of this sentence](https://en.wikipedia.org/wiki/Gerund#Gerunds_in_English), which makes more sense. Then the "would" to "does" here change follows to match that.

Reply:
Done.